dht/dhtcore/NodeStore.c

```diff
@@ -1001,6 +1001,8 @@ static struct Node_Link* discoverLinkC(struct NodeStore_pvt* store,
     }
 
     struct Node_Two* parent = closest->child;
+    uint64_t realPathParentChild = pathParentChild == 1 ?
+        closest->cannonicalLabel : pathParentChild;
```

Comment:
Technically ```if (discoveredPath == child->address.path) {``` should be enough because it will eliminate any switches not in the best path to the child from causing that child to be unlinked. Also there is a possibility to do this synchronously, IE to change closest to Node_getBestParent(closest->parent) and then continue on rather than returning NULL, this will save 1 RTT in correcting the link.